src/amp-story-player/amp-story-player-impl.js

```diff
@@ -833,17 +863,23 @@ export class AmpStoryPlayer {
    * Gets encoded url for player usage.
    * @param {string} href
    * @param {VisibilityState=} visibilityState
+   * @param {EmbedMode=} embedMode
    * @return {!Location}
    * @private
    */
-  getEncodedLocation_(href, visibilityState = VisibilityState.INACTIVE) {
+  getEncodedLocation_(
+    href,
+    visibilityState = VisibilityState.INACTIVE,
+    embedMode = EmbedMode.NOT_EMBEDDED
+  ) {
     const params = dict({
       'amp_js_v': '0.1',
       'visibilityState': visibilityState,
       'origin': this.win_.origin,
       'showStoryUrlInfo': '0',
       'storyPlayer': 'v0',
       'cap': 'swipe',
+      [EmbedModeParam]: embedMode,
```

Comment:
I think we need to take into account the case where the publisher already specified an `embedMode`. 

e.g. there are some users of the player already setting the stories in their player like this:
```
<amp-story-player>
  <a href="mystory.com/story.html#embedMode=1"> // We should respect embedMode=1
</amp-story-player>
```

Reply:
I tried testing this in master (without the changes in this PR) and noticed the current bahavior may be incorrect anyways.

I added the following story: `https://www-usatoday-com.cdn.ampproject.org/c/s/www.usatoday.com/amp-stories/beyonce-rule-the-world/#embedMode=3`

It already has a fragment, as you can see. When parsed with this method, the output is: `https://www-usatoday-com.cdn.ampproject.org/c/s/www.usatoday.com/amp-stories/beyonce-rule-the-world/#embedMode=3?amp_js_v=0.1#visibilityState=prerender&origin=http%3A%2F%2Flocalhost%3A8000&showStoryUrlInfo=0&storyPlayer=v0&cap=swipe`

This has two fragment parameter sections, which is an invalid URL. I am actually not sure how it ends up working, but I think I'm going to end up having to change a significant part of this method.

Reply:
With further testing, I found that, in that last URL

```
https://www-usatoday-com.cdn.ampproject.org/c/s/www.usatoday.com/amp-stories/beyonce-rule-the-world/#embedMode=3?amp_js_v=0.1#visibilityState=prerender&origin=http%3A%2F%2Flocalhost%3A8000&showStoryUrlInfo=0&storyPlayer=v0&cap=swipe
```

The hash part is considered to be:

```
#embedMode=3?amp_js_v=0.1#visibilityState=prerender&origin=http%3A%2F%2Flocalhost%3A8000&showStoryUrlInfo=0&storyPlayer=v0&cap=swipe
```

With the `parseQueryString` function from `src/url`, this is parsed as:

```
{
  cap: "swipe"
  embedMode: "3?amp_js_v=0.1#visibilityState=prerender"
  origin: "http://localhost:8000"
  showStoryUrlInfo: "0"
  storyPlayer: "v0"
}
```

Notice how the `amp_js_v` query and the `visibilityState` fragment are all grouped under the `embedMode` parameter. This still results in the appropiate embedMode being applied, as the `parseEmbedMode` function [here](https://github.com/ampproject/amphtml/blob/2fb6f8e0d7ebb0a0944faf91c1a8f306db8d9146/extensions/amp-story/1.0/embed-mode.js#L91) uses `parseInt`, which parses integers until non-integers are found, still giving 3 as a result, appropiately.

I have also found that some stories do actually break with an incorrect URL (missing `amp_js_v` query), like [this one](https://www-washingtonpost-com.cdn.ampproject.org/v/s/www.washingtonpost.com/graphics/2019/lifestyle/travel/amp-stories/a-locals-guide-to-what-to-eat-and-do-in-miami/#embedMode=3?amp_js_v=0.1#visibilityState=prerender&origin=http%3A%2F%2Flocalhost%3A8000&showStoryUrlInfo=0&storyPlayer=v0&cap=swipe).

I'm fixing this in the next commits.

Reply:
After I uploaded the "fixes", I noticed a test for this started failing. Turns out it was checking against the following URL:
```
https://www-washingtonpost-com.cdn.ampproject.org/v/s/www.washingtonpost.com/graphics/2019/lifestyle/travel/amp-stories/a-locals-guide-to-what-to-eat-and-do-in-new-york-city/?testParam=true#myhash=hashValue&amp_js_v=0.1#visibilityState=visible&origin=http%3A%2F%2Flocalhost%3A9876&showStoryUrlInfo=0&storyPlayer=v0&cap=swipe&embedMode=0
```
This is also incorrect (has 2 hashes), so I changed it.

Reply:
This is great! Thanks for catching and investigating this so thoroughly. 👏👏👏